Change rbac version to "v1alpha1".

liminw opened the pull request.

liminw commented:
cc @wora

istio-testing commented:
@liminw: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/api-presubmit.sh | 81406a5620a69943ef2e72c024bbbf78f207711e | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_api/296/api-presubmit/203/) | `/test api-presubmit`



<details>

Instructions for interacting with me using PR comments are available [here](https://github.com/kubernetes/community/blob/master/contributors/devel/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://github.com/kubernetes/test-infra/blob/master/commands.md).
</details>
<!-- test report -->

wora commented on rbac/v1alpha1/rbac.proto:
```diff
@@ -32,7 +32,7 @@ package istio.rbac.v1;
 //
-//     apiVersion: "config.istio.io/v1alpha2"
+//     apiVersion: "config.istio.io/v1alpha1"
 //     kind: authorization
```
is kind title case or lower case?

liminw replied:
I am going to change kind to CamelCase "Authorization". It seems that
* Kubernetes uses CamelCase for kind. For example, "RoleBinding", "ClusterRole". 
* Pilot also uses CamelCase. For example, "RouteRule". 
* Mixer uses lower case. For example, "attributemanifest", "listchecker". 

We should fix the inconsistency. @geeknoid @mandarjog

liminw commented:
/retest

istio-testing commented:
@liminw: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/api-presubmit.sh | 81406a5620a69943ef2e72c024bbbf78f207711e | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_api/296/api-presubmit/220/) | `/test api-presubmit`



<details>

Instructions for interacting with me using PR comments are available [here](https://github.com/kubernetes/community/blob/master/contributors/devel/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://github.com/kubernetes/test-infra/blob/master/commands.md).
</details>
<!-- test report -->

kyessenov commented:
@geeknoid 
It seems that there are some comments, and the API might need a few iterations to be included in Istio. Do you think it would be a good idea to continue working on a branch before merging into master? It's difficult to quickly iterate on an API once istio has some code that uses it, since it leads to repos going out of sync, and blocking subsequent PRs to istio/api.

geeknoid commented:
Limin, what do you think? Can you address Hong's questions, or should we
move this to a branch for further bake time?

On Thu, Jan 11, 2018 at 11:48 AM, Kuat <notifications@github.com> wrote:

> @geeknoid <https://github.com/geeknoid>
> It seems that there are some comments, and the API might need a few
> iterations to be included in Istio. Do you think it would be a good idea to
> continue working on a branch before merging into master? It's difficult to
> quickly iterate on an API once istio has some code that uses it, since it
> leads to repos going out of sync, and blocking subsequent PRs to istio/api.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/istio/api/pull/296#issuecomment-357040759>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AVucHaCIBZ6z8RNJTOheZKGJ91m8Vgo5ks5tJmV8gaJpZM4RYqma>
> .
>

liminw commented:
@wora Sorry for replying late. 

gRPC has two types, some supports HTTP transcoding, some do not (i.e., native gRPC). You mentioned the case that gRPC supports HTTP transcoding, but we also need to support native gRPC calls. 

"constraints" limits the permission to a certain condition. For example, if you have a constraint "version: v1", it means that the permission only applies to services with "v1" version. If you think that is not a good name, I can rename it to "properties" or something else. 

Properties are not labels or tags, they are mapped from attributes. Calling them labels or tags will be confusing. 

I am not sure what you mean "k8s has standard way to refer resources". k8s RBAC also has "roleRef" referring to roles. Here is an example of k8s RBAC. 

kind: RoleBinding
apiVersion: rbac.authorization.k8s.io/v1
metadata:
  name: read-pods
  namespace: default
subjects:
- kind: User
  name: jane
  apiGroup: rbac.authorization.k8s.io
roleRef:
  kind: Role
  name: pod-reader
  apiGroup: rbac.authorization.k8s.io

kyessenov commented:
Yeah, we should use camel case. Just because k8s is lenient now, doesn't
mean it's going to stay that way.

On Fri, Jan 12, 2018 at 12:58 PM Limin Wang <notifications@github.com>
wrote:

> *@liminw* commented on this pull request.
> ------------------------------
>
> In rbac/v1alpha1/rbac.proto
> <https://github.com/istio/api/pull/296#discussion_r161319824>:
>
> > @@ -32,7 +32,7 @@ package istio.rbac.v1;
>  //
>  // For example, suppose we define an instance of the "authorization" template, named "requestcontext".
>  //
> -//     apiVersion: "config.istio.io/v1alpha2"
> +//     apiVersion: "config.istio.io/v1alpha1"
>  //     kind: authorization
>
> I am going to change kind to CamelCase "Authorization". It seems that
>
>    - Kubernetes uses CamelCase for kind. For example, "RoleBinding",
>    "ClusterRole".
>    - Pilot also uses CamelCase. For example, "RouteRule".
>    - Mixer uses lower case. For example, "attributemanifest",
>    "listchecker".
>
> We should fix the inconsistency. @geeknoid <https://github.com/geeknoid>
> @mandarjog <https://github.com/mandarjog>
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/istio/api/pull/296#discussion_r161319824>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AJGIxrM-w6LO_ZFzzozimc-rx3zHBsU5ks5tJ8PpgaJpZM4RYqma>
> .
>

wora commented:
I chatted with @liminw in person yesterday. I think we have reached an agreement. She will make some changes based on our discussion.

liminw commented:
@wora I have addressed your comments in the new commits.

@geeknoid @kyessenov I filed Issue #2633 to track CRD kind CamelCase vs. lowercase problem.
